docs/guides/usage.md

```diff
@@ -0,0 +1,83 @@
+# How to use the lakeFS file system
+
+This guide contains instructions and code snippets on how to use the lakeFS file system.
+
+## Uploading and downloading files
+
+The arguably most important feature of the file system are file transfers.
+
+### File uploads
+
+To upload a file, you can use the `fs.put()` and `fs.put_file` methods:
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+# remote path, then local target path.
+fs.put_file("file.txt", "my-repo/my-ref/file.txt")
+```
+
+If you want to upload a whole directory to lakeFS, use the `fs.put` API together with the `recursive=True` switch:
```

Comment:
```suggestion
If you want to upload an entire directory to lakeFS, you can use the `fs.put()` API together with the `recursive=True` switch:
```

Reply:
Changed all occurrences of `whole` to `entire` in [13917c0](https://github.com/aai-institute/lakefs-spec/pull/156/commits/13917c064afc3816ce4cca9787ebb3ad784d3c98).